corehq/apps/geospatial/views.py

```diff
@@ -171,3 +189,124 @@ def post(self, request, *args, **kwargs):
         instance.save()
 
         return self.get(request, *args, **kwargs)
+
+
+class GPSCaptureView(BaseDomainView):
+    urlname = 'gps_capture'
+    template_name = 'gps_capture.html'
+
+    page_name = _("Manage GPS Data")
+    section_name = _("Geospatial")
+
+    @method_decorator(toggles.GEOSPATIAL.required_decorator())
+    def dispatch(self, *args, **kwargs):
+        return super(GPSCaptureView, self).dispatch(*args, **kwargs)
+
+    @property
+    def section_url(self):
+        return reverse(self.urlname, args=(self.domain,))
+
+    @property
+    def page_url(self):
+        return reverse(self.urlname, args=(self.domain,))
+
+    @property
+    def page_context(self):
+        data_type = self.request.GET.get('data_type', 'case')
+        return {
+            'data_type': data_type
+        }
+
+    @method_decorator(toggles.GEOSPATIAL.required_decorator())
+    def post(self, request, *args, **kwargs):
+        json_data = json.loads(request.body)
+        data_type = json_data.get('data_type', None)
+        data_item = json_data.get('data_item', None)
+
+        if data_type == 'case':
+            process_gps_values_for_case(request.domain, data_item)
+        elif data_type == 'user':
+            process_gps_values_for_user(request.domain, data_item)
+
+        return json_response({
+            'status': 'success'
+        })
+
+
+@require_GET
+@login_and_domain_required
+def get_paginated_cases_or_users_without_gps(request, domain):
+    page = max(int(request.GET.get('page', 1)), 1)
+    limit = max(int(request.GET.get('limit', 5)), 1)
+    query = request.GET.get('query', '')
+    case_or_user = request.GET.get('data_type', 'case')
+    if case_or_user not in ['case', 'user']:
```

Comment:
Should we not rather raise an exception here? Feels like this can lead to unexpected/confusing behaviour when `data_type` is neither `case` nor `user`. If for some reason we do receive a `data_type` other then `case` or `user` (which would happen when someone fiddles with something they should not or during development), we can just raise an error and let them know that the request is invalid. Maybe a [400 status](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400) or something

Reply:
There were originally asserts in the view, but these have since been changed after [this](https://github.com/dimagi/commcare-hq/pull/33420#discussion_r1318438071) comment.

What makes me hesitant for returning an exception is that the user might navigate to the GPS capture page without specifying the URL param. In this scenario no data would load and they would see an error banner. Perhaps this outcome is fine when the user tries to just copy/paste the URL?

Alternatively, we could also add a `message` in the response JSON so for example if the wrong `data_type` was specified then we return with something like "<data_type> is not valid. Defaulting to 'case'>".

Reply:
From [the comment](https://github.com/dimagi/commcare-hq/pull/33420#discussion_r1318438071) you linked it looks like we handle these situations in two different ways in the codebase i.e. 500 and default to some value.
From this:
> For the capture page, if these assertions managed to fail then a 500 will be returned and the user will see an error banner.

I like the idea of telling the user what went wrong, which is why I'd personally prefer raising an exception or something. That said, since 1. this situation will only happen during development or if the user tampered with the URL (which should not happen), and 2. we control what's being sent to the endpoint, we can technically chuck these checks and assume it to be correct? If the user tampers with the URL, they should expect there to be dragons (it will probably 500 somewhere).

All-in-all, I'm sort of gonna go back on my statement earlier. Doing checks here for something we control and know will be correct feels unncessary.

Reply:
> In this scenario no data would load and they would see an error banner.

This is fine?

Reply:
> Doing checks here for something we control and know will be correct feels unncessary

This is a good point, we know where this endpoint is going to be used and can control what we're sending to it. In that case, I agree these checks could be removed and take a similar approach to what we're doing for retrieving paginated attendees in attendance tracking.

Adding on to the above, I know we'll also be looking to remove the `data_type` param in the future and replace it with UI tabs where users can switch between cases/users. So I think given this and the point above I'll remove the checks.